test/pmi_client.c

```diff
@@ -334,30 +328,6 @@ static int test_item5(void)
 static int test_item6(void)
 {
     int rc = 0;
-    char val[100];
-    const char *tkey = __func__;
-    const char *tval = __FILE__;
-
-    if (PMI_SUCCESS != (rc = PMI_KVS_Put(jobid, tkey, tval))) {
-        log_fatal("PMI_KVS_Put %d\n", rc);
-        return rc;
-    }
-
-    if (PMI_SUCCESS != (rc = PMI_KVS_Get(jobid, tkey, val, sizeof(val)))) {
```

Comment:
It's a fair question. IIRC, we created "store_internal" to specifically allow storing of data that would not be shared to other procs. This data is still retrieved using PMIx_Get, and so the Get call must check the internal hash table when given a proc's own name.

Given that we have to do the check anyway, there is no reason not to have a proc's own data be immediately available - we just always store a copy in the local hash table. If we do that, then it probably makes more sense to remove the Store_internal API and just use PMIx_Put with PMIX_SCOPE_LOCAL for the "do not share" case instead.